Sources/SwiftToolchain/ToolchainManagement.swift

```diff
@@ -166,6 +167,12 @@ public class ToolchainSystem {
       }.whenComplete($0)
     }
 
+    #if arch(x86_64)
+    let archSuffix = "x86_64"
+    #elseif arch(arm64)
+    let archSuffix = "arm64"
+    #endif
```

Comment:
(not big deal)
FYI: `NXGetLocalArchInfo` is more generic way to get the current architecture mode.
You can see how to use it from `arch` command source https://opensource.apple.com/source/cctools/cctools-886/libmacho/arch.c.auto.html

Reply:
This won't work on Linux, right?

Reply:
Yes, Linux has another mechanism called sysinfo

Reply:
Right, but what would be the benefit of using `sysinfo` on Linux, `NXGetLocalArchInfo` on macOS, and then maybe some third mechanism for Windows (if we ever get there), while `#if arch` seems to achieve the same. Or maybe I'm missing something?

Reply:
If a user uses a carton built for x64 on arm64 darwin, it downloads x64 toolchain in current implementation. If we get the architecture at runtime, we can avoid such situation and downloads arm64 toolchain even on Rosetta. But I'm not confident which behavior is natural.

Reply:
I'm not sure if there's much benefit in allow this configuration. I still want to be able to download and test x86_64 toolchains on my M1 hardware. Natural way to do it right now is to just run `carton` under Rosetta. Otherwise we'd have to come up with some CLI flags to make it configurable, which is probably not worth it? I'm ready to change my opinion on this though if a good use case for making this configurable appears.